llvm/lib/CodeGen/SelectOptimize.cpp

```diff
@@ -177,11 +189,22 @@ class SelectOptimizeImpl {
       llvm_unreachable("Unhandled case in getCondition");
     }
 
+    /// Return the condition for the SelectLike instruction. For example the
+    /// condition of a select or c in `or(zext(c), x)`
+    Value *getCondition() const {
+      Value *CC = getNonInvertedCondition();
+      if (Inverted)
+        return cast<Instruction>(CC)->getOperand(0);
```

Comment:
add a comment on what type of instruction can be inverted and why it is safe/correct to get the first operand for the condition